can-component.js

```diff
@@ -161,6 +162,12 @@ var Component = Construct.extend(
 					this.Control = ComponentControl.extend(this.prototype.events);
 				}
 
+				//!steal-remove-start
+				// If a constructor is assigned to the viewModel, give a warning
+				if (this.prototype.viewModel && canReflect.isConstructorLike(this.prototype.viewModel)) {
+					canDev.warn("Assigning a DefineMap or constructor type to the viewModel property may not be what you intended. Did you mean ViewModel instead? More info: https://canjs.com/doc/can-component.prototype.ViewModel.html");
```

Comment:
I think we should always include which package is warning.  Include a `can-component: Assigning a ...` in the warning.